tests/python_package_test/test_basic.py

```diff
@@ -621,3 +622,32 @@ def test_no_copy_when_single_float_dtype_dataframe(dtype):
     built_data = lgb.basic._data_from_pandas(df, feature_name, None, None)[0]
     assert built_data.dtype == dtype
     assert np.shares_memory(X, built_data)
+
+
+@pytest.mark.parametrize('min_data_in_bin', [2, 10])
+def test_feature_num_bin(min_data_in_bin):
+    X = np.vstack([
+        np.random.rand(100),
+        np.array([1, 2] * 50),
+        np.array([0, 1, 2] * 33 + [0]),
+        np.array([1, 2] * 49 + 2 * [np.nan]),
+        np.zeros(100),
+    ]).T
+    ds = lgb.Dataset(X, params={'min_data_in_bin': min_data_in_bin}).construct()
+    expected_num_bins = np.array([
+        ceil(100 / min_data_in_bin) + 1,  # extra bin for zero
+        3,  # 0, 1, 2
+        3,  # 0, 1, 2
+        4,  # 0, 1, 2 + nan
+        0,  # unused
+    ])
+    actual_num_bins = [ds.feature_num_bin(i) for i in range(X.shape[1])]
+    np.testing.assert_equal(actual_num_bins, expected_num_bins)
```

Comment:
I think we can simply assert two lists here.
```suggestion
    expected_num_bins = [
        ceil(100 / min_data_in_bin) + 1,  # extra bin for zero
        3,  # 0, 1, 2
        3,  # 0, 1, 2
        4,  # 0, 1, 2 + nan
        0,  # unused
    ]
    actual_num_bins = [ds.feature_num_bin(i) for i in range(X.shape[1])]
    assert actual_num_bins == expected_num_bins
```